gobblin-compaction/src/main/java/gobblin/compaction/parser/CompactionParser.java

```diff
@@ -0,0 +1,15 @@
+package gobblin.compaction.parser;
+
+import gobblin.configuration.State;
+import gobblin.dataset.Dataset;
+import lombok.AllArgsConstructor;
+
+/**
+ * A parser class which can convert a given {@link Dataset} to any object defined by user
+ */
+@AllArgsConstructor
+public abstract class CompactionParser<D extends Dataset> {
```

Comment:
What is the use of this as an abstract class? Can you give me an example of a different `CompactionParser` we might want in the future?